src/runtime/HalideRuntime.h

```diff
@@ -1334,6 +1334,7 @@ typedef enum halide_target_feature_t {
     halide_target_feature_egl,                    ///< Force use of EGL support.
     halide_target_feature_arm_dot_prod,           ///< Enable ARMv8.2-a dotprod extension (i.e. udot and sdot instructions)
     halide_llvm_large_code_model,                 ///< Use the LLVM large code model to compile
+    halide_target_feature_avx512_sapphirerapids,  ///< Enable the AVX512 features supported by Sapphire Rapids processors. This include all of the Cannonlake features, plus AVX512-VNNI and AVX512-BF16.
```

Comment:
Comment #1: this should probably just be grouped with the other avx512 features earlier in the list. (Note that Halide feature enums are *not* promised to be identical across Halide versions, so there is no need to add them at the end.) Probably ditto for grouping in the other places it appears in this PR.

Comment #2: I realize the wording of "includes all of the $FOO features" appears in the comments for the existing AVX512 features, but is it really true? i.e., does specifying "avx512_sapphirerapids" in a target string really implicitly select all the earlier features? (I don't see any code in Halide proper that seems to enforce this; does the `mcpu()` string we hand to LLVM end up enforcing this?)

Reply:
For 2 that's a good point. There is some code in [CodeGen X86](https://github.com/halide/Halide/blob/3defb66236c3cd6812db90a651ff13f6f8170651/src/CodeGen_X86.cpp#L29) that partially addresses this. I should add to that to include SapphireRapids.